src/lib/Libwin/passwd.c

```diff
@@ -2688,23 +2688,26 @@ getHomedir(char *user)
 	char	*homedir = NULL;
 	HANDLE	userlogin = INVALID_HANDLE_VALUE;
 	int	i;
+	struct passwd *pwdp;
 
 
 	homedir = getAssignedHomeDirectory(user);
 
 	if (homedir)
 		return (homedir);
 
-	/* look for a user login handle */
-	for (i=0; i<user_num; i++) {
-		struct passwd *pwdp;
-
-		pwdp = &user_array[i];
+	if(user_num == 0){
```

Comment:
Since it is a linked list of user password structures anyway, why do you need the user_num variable at all?